### PR TITLE
[AR-4705] Add support wallet watch asset

### PR DESCRIPTION
**Pull request by karthik-durai**
# PR

## Describe your changes
1. Added logic and UI to send request to wallet to add token.

## Issue ticket number and link

- [AR-4705](https://team-1624093970686.atlassian.net/browse/AR-4705)

## Checklist before requesting a review

- [X] You have performed a self-review of your own code
- [X] You are using approved terminology
- [X] Your changes have been tested locally
- [X] Your code builds clean without any errors or warnings

### CODE REVIEW

File: examples/index.js [hunk 1]
```diff
@@ -93,6 +93,21 @@ async function switchChain() {
   }
 }
 
+async function addToken() {
```
Comment (makylfang) [inline examples/index.js:96]:
```
ethereum.request({
  method: 'wallet_watchAsset',
  params: {
    type: 'ERC20',
    options: {
      address: '0xb60e8dd61c5d32be8058bb8eb970870f07233155',
      symbol: 'FOO',
      decimals: 18,
      image: 'https://foo.io/token-image.svg',
    },
  },
});
```
It's like this according to EIP.

Comment (makylfang) [reply examples/index.js:96]: The address will be in options key, and params will be an object.

Comment (karthik-durai) [reply examples/index.js:96]: https://github.com/arcana-network/auth/pull/127/commits/3b7d875e916076f704692a66cf1ec3794762f476

**Review by ajithranka — APPROVED** [review]

**Comment by makylfang** [comment]
The `params` has to be an object and not array of objects.

```js
await provider.request({
  method: 'wallet_watchAsset',
  params: {
    type: 'ERC20',
    options: {
      address: '0xB983E01458529665007fF7E0CDdeCDB74B967Eb6',
      symbol: 'FOO',
      decimals: 18,
      image: 'https://foo.io/token-image.svg',
    },
  },
});
```

instead of 

```js
await provider.request({
  method: 'wallet_watchAsset',
  params: [
    {
      type: 'ERC20',
      options: {
        address: '0xB983E01458529665007fF7E0CDdeCDB74B967Eb6',
        symbol: 'FOO',
        decimals: 18,
        image: 'https://foo.io/token-image.svg',
      },
    },
  ],
});
```

@karthik-durai

**Review by makylfang — APPROVED** [review]

**Comment by karthik-durai** [comment]
> The `params` has to be an object and not array of objects.
> 
> ```js
> await provider.request({
>   method: 'wallet_watchAsset',
>   params: {
>     type: 'ERC20',
>     options: {
>       address: '0xB983E01458529665007fF7E0CDdeCDB74B967Eb6',
>       symbol: 'FOO',
>       decimals: 18,
>       image: 'https://foo.io/token-image.svg',
>     },
>   },
> });
> ```
> 
> instead of
> 
> ```js
> await provider.request({
>   method: 'wallet_watchAsset',
>   params: [
>     {
>       type: 'ERC20',
>       options: {
>         address: '0xB983E01458529665007fF7E0CDdeCDB74B967Eb6',
>         symbol: 'FOO',
>         decimals: 18,
>         image: 'https://foo.io/token-image.svg',
>       },
>     },
>   ],
> });
> ```
> 
> @karthik-durai

@makylfang  https://github.com/arcana-network/auth/pull/127/files#r1032038867

**Comment by makylfang** [comment]
@karthik-durai Merge?

**Pull request merged**